services/web/frontend/js/lda.js

```diff
@@ -68,15 +68,17 @@ $(function() {
                             console.log('success in training file POST');
                             $('#tm-success-message').removeAttr('hidden');
                         },
-                        error: function (err) {
-                            console.log(err);
-                            $('#err-uploading').removeAttr('hidden');
+                        error: function (xhr, status, err) {
+                            console.log(xhr.responseText);
+                            $('#err-uploading').html(`An error occurred while uploading your file: ${JSON.parse(xhr.responseText).message}`)
```

Comment:
So, it turns out, there are two kinds of error messages that the server sends. The first format (which you've addressed) is:

```javascript
{
   "message": "...",
}
```
The second format is
```javascript
{
    "message": {
        "num_topics": "The number of topics must be an integer greater than 1."
    }
}
```

The second format is returned only when the error pertains specifically to one field (like the number of topics, in this example). 

Could you try to account for both types of errors here (and also on the classifiers page)?

Tbh, the possible errors the server can spew out should be documented in the backend API. I'll create an issue for that.

Reply:
Will do, where can I find the other formats that might get returned? Or is num_topics the only nested one?